### PR TITLE
Add options from Google API

### PR DESCRIPTION
**Pull request by falkodev**

### CODE REVIEW

File: views/widget.html [hunk 1]
```diff
@@ -2,7 +2,15 @@
 {% set id = apos.utils.generateId() %}
 {% set widget = data.widget %}
 {% set classPrefix = data.widget.classPrefix %}
-<script src="https://maps.google.com/maps/api/js?key={{ widget.googleApiKey }}&libraries=places" type="text/javascript"></script>
+<script src="https://maps.google.com/maps/api/js?key={{ widget.googleApiKey }}&libraries=places
```
Comment (abea) [inline views/widget.html]: We should include `defer` on this. Or use `apos.templates.append('body', ...)` to put it at the end of the DOM.

File: index.js [hunk 2]
```diff
@@ -24,7 +361,14 @@ module.exports = {
         'fieldName',
         'required',
         'googleApiKey',
-        'placeholder'
+        'placeholder',
+        'types',
```
Comment (abea) [inline index.js]: I think these advanced options should be in a separate "Advanced" field tab.

**Review by abea — CHANGES_REQUESTED** [review]

File: views/widget.html (hunk 1 above)
Comment (falkodev) [reply views/widget.html]: Nice for `defer`. Should I add `async` too?

For `apos.templates.append` how would you use it? I tried it but did not understand how to get widget values.

File: index.js (hunk 2 above)
Comment (falkodev) [reply index.js]: Good idea.

File: README.md [hunk 3]
```diff
@@ -14,8 +14,8 @@ modules: {
     }
   },
   'apostrophe-forms-widgets': {},
-  'apostrophe-forms-google-address-field-widgets'
+  'apostrophe-forms-google-address-field-widgets': {}
 }
 ```
 
-The user will need a valid Google API key for the suggestions displayed by Google to work.
+The user will need a valid Google API key for the suggestions displayed by Google to work as stated in Google Maps Platform documentation for Places API: [https://developers.google.com/places/web-service/autocomplete](https://developers.google.com/places/web-service/autocomplete) A premium plan is needed to obtain a key from Google.
```
Comment (abea) [inline README.md]: Is billing required? Or is it only required if you hit a certain limit? I think you are simply limited to lower usage without billing.

**Review by abea — CHANGES_REQUESTED** [review]
I'm getting some more errors. I think the Google script should be added in the `lean.js` file, and only if `window.google` does not already exist, to make sure it's not loaded more than once. [The second error links here](https://developers.google.com/maps/documentation/javascript/error-messages#api-not-activated-map-error). I have the Places API active, but not the Maps API. Maybe that's required as well?

Screen Shot 2020-06-25 at 9 38 59 AM

The second error is obscured, but it is the same as the first.

File: README.md (hunk 3 above)
Comment (falkodev) [reply README.md]: It is yes. They offer you $200 when subscribing though.

**Comment by falkodev** [comment]
> I'm getting some more errors. I think the Google script should be added in the `lean.js` file, and only if `window.google` does not already exist, to make sure it's not loaded more than once. [The second error links here](https://developers.google.com/maps/documentation/javascript/error-messages#api-not-activated-map-error). I have the Places API active, but not the Maps API. Maybe that's required as well?
> 
> Screen Shot 2020-06-25 at 9 38 59 AM
> 
> The second error is obscured, but it is the same as the first.

Done. The script is loaded through `lean.js` now and it is loaded only once for the page.

File: index.js [hunk 4]
```diff
@@ -6,26 +6,1159 @@ module.exports = {
...
+      type: 'array',
+      titleField: 'country',
+      limit: 5,
+      help: 'Restrict search to one or multiple countries (5 at max due to Google restriction). Might be not compatible with origin or bounds values.',
+      schema: [
+        {
+          name: 'country',
+          label: 'Country',
+          type: 'select',
+          choices: [
```
Comment (abea) [inline index.js]: I suggest moving this array into it's own file and import it here to simply make this file more manageable.

File: README.md (hunk 3 above)
Comment (abea) [reply README.md]: It seems like this requires the Maps Javascript API as well. We should be very clear which need to be enabled.

File: index.js [hunk 5]
```diff
@@ -6,26 +6,1159 @@ module.exports = {
...
-      fields: [
-        'fieldLabel',
-        'fieldName',
-        'required',
-        'googleApiKey',
-        'placeholder'
+    },
+    {
+      name: 'types',
+      label: 'Type',
```
Comment (abea) [inline index.js]: "Google Places Autocomplete Type" would be clearer.

File: README.md (hunk 3 above)
Comment (abea) [reply README.md]: It seems like this requires the Maps Javascript API as well. We should be very clear which need to be enabled.

I don't think Google phrases it like "premium plan." I suggest phrasing it as saying something like "you must connect a billing account to the Google API console project."

**Review by abea — CHANGES_REQUESTED** [review]
It works nicely. I have a few notes about documentation and wording. And I suggest moving the countries to a separate file to make index.js more manageable, but that's optional.

File: README.md (hunk 3 above)
Comment (falkodev) [reply README.md]: Thanks for the phrasing

File: index.js (hunk 4 above)
Comment (falkodev) [reply index.js]: Done

File: index.js (hunk 5 above)
Comment (falkodev) [reply index.js]: Done

**Review by abea — APPROVED** [review]

**Pull request merged**